src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -295,7 +295,10 @@ protected override int MakeSlot(BoundExpression node)
             return base.MakeSlot(node);
         }
 
-        private new void VisitLvalue(BoundExpression node)
+        /// <summary>
+        /// If you will be needing AssignedWhenTrue and AssignedWhenFalse, set `keepSplit` to `true`.
+        /// </summary>
+        private void VisitLvalue(BoundExpression node, bool keepSplit = false)
```

Comment:
When is `keepSplit: true` for an l-value?

Reply:
In `VisitArgumentEvaluate` when the ref kind is `out` for example.

---
In reply to: [192465175](https://github.com/dotnet/roslyn/pull/27313#discussion_r192465175) [](ancestors = 192465175)

Reply:
From our discussion there is no practical scenario for this. Simplified in last commit. Thanks

---
In reply to: [193156697](https://github.com/dotnet/roslyn/pull/27313#discussion_r193156697) [](ancestors = 193156697,192465175)